framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -128,6 +158,29 @@ PatternedMeshGenerator::generate()
   if (_z_width == 0)
     _z_width = bbox.max()(2) - bbox.min()(2);
 
+  std::vector<boundary_id_type> stitch_bids;
+  if (have_common_ids)
+    stitch_bids = input_bids[0];
+  else
+  {
+    for (boundary_id_type id = 0; id != Moose::INVALID_BOUNDARY_ID; ++id)
+      if (!used_boundary_ids.count(id))
+      {
+        stitch_bids.push_back(id);
+        if (stitch_bids.size() == 4)
+          break;
+      }
+
+    // Make all inputs have common boundary ids
+    for (const auto i : index_range(_meshes))
+      for (const auto side : make_range(4))
+        MeshTools::Modification::change_boundary_id(
+            *_meshes[i], input_bids[i][side], stitch_bids[side]);
```

Comment:
this could be problematic if input_bids[i] is the same on all sides.

Then you end up with a no sideset with stitch_bids[0, 1 or 2]

Reply:
Not possible by default: https://github.com/idaholab/moose/blob/470e4353ecbd7163dd55ed236bcd81eb3cc485ba/framework/src/meshgenerators/PatternedMeshGenerator.C#L91

We even discussed this (this error is what caused problems with reactor)

why must you hijack my PR

Reply:
no I think it breaks as I say
```
[Mesh]
  [gmg]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 5
    ny = 5
    subdomain_ids = '1 1 1 1 1
                     1 1 1 1 1
                     1 1 1 1 1
                     1 1 1 1 1
                     1 1 1 1 1
                     '
  []

  [gmg2]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 5
    ny = 5
    subdomain_ids = '2 2 2 2 2
                     2 2 2 2 2
                     2 2 2 2 2
                     2 2 2 2 2
                     2 2 2 2 2
                     '
    # The following triggers generation of new common boundary ids in
    # PatternedMeshGenerator
    boundary_id_offset = 1
  []

  [2_sides_same]
    type = RenameBoundaryGenerator
    input = gmg2
    old_boundary = 'left top'
    new_boundary = 'left left'
  []
  [2_sides_same2]
    type = RenameBoundaryGenerator
    input = gmg
    old_boundary = 'left top'
    new_boundary = 'left left'
  []

  [pmg]
    type = PatternedMeshGenerator
    inputs = '2_sides_same2 2_sides_same'
    pattern = '1 0 ;
               0 1'
    top_boundary = 'left'
    require_unique_boundaries = false
  []
[]
```

```
In UnstructuredMesh::stitch_meshes:
This mesh has 11 nodes on boundary 8.
Other mesh has 0 nodes on boundary 7.
Minimum edge length on both surfaces is 0.2.
In UnstructuredMesh::stitch_meshes:
Found 0 matching nodes.
```

Reply:
Setting 'require_unique_boundaries' to true (the default value) fixes this issue ;)

Reply:
unless you forbid users from setting it to false then there's an issue.

It fails silently too which is more problematic